### DISCUSSION TITLE
My code skips 'nested if statements' and runs to infinity when using a Webcam.

### DISCUSSION THREAD

**Discussion by cruseruka** [ultralytics/ultralytics#2724] [Q&A]
Hi everyone. I'm using a YOLOv5 tracking code. I've modified it, it fails to perform some tasks:
1. Runs to infinity (when I opt to use a webcam, it doesn't accept the WaitKey() function),
2. I gave it some conditions to check before it does some detections and labeling. See line 177, I want it to detect a bottle first, then perform other detections (cup, line 197; book, line 214; and cell phone, line 223).
Unfortunately, when I show it a bottle, it detects, annotates, and labels it, but skips the other nested 'if' statements.

I have attached the code and the weights I'm using.

Kindly help.
[Code+Weights.zip](https://github.com/ultralytics/ultralytics/files/11523642/Code%2BWeights.zip)

![image](https://github.com/ultralytics/ultralytics/assets/98976993/25afc373-ed53-4808-ae78-e020dd0ec1bf)

**Comment by glenn-jocher** [comment]
@cruseruka hello! It sounds like you're experiencing a couple of issues with your modified YOLOv5 tracking code. Since you're working with YOLOv5, I would recommend checking the YOLOv5 repository and its issues section for more specific guidance related to that codebase.

However, if you're interested in transitioning to YOLOv8, which is the focus of this repository, you'll find that it offers robust tracking capabilities and may provide improved performance or easier debugging for your application. YOLOv8 is designed to handle real-time object tracking, and you can utilize the Track mode for this purpose.

For the infinite loop issue, it's typically related to the logic handling the webcam stream and the `cv2.waitKey()` function. Ensure that you're using the correct condition to break out of the loop when a specific key is pressed.

Regarding the nested 'if' statements, it's crucial to ensure that the logic is structured correctly to allow for sequential detections. In YOLOv8, you can handle multiple object detections in a single pass, and the model will annotate and label all detected objects according to the confidence thresholds set.

If you decide to use YOLOv8 and encounter similar issues, please refer to the documentation at https://docs.ultralytics.com for guidance on using the various modes and tasks that YOLOv8 supports. If you still face difficulties, feel free to open an issue in the YOLOv8 repository with a detailed description of the problem, and we'll be happy to assist you.

Remember, the YOLO community and the Ultralytics team are here to support you. Good luck with your object tracking project! 🚀
